src/audio/codec_adapter/codec/generic.c

```diff
@@ -0,0 +1,359 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+
+/*
+ * \file generic.c
+ * \brief Generic Codec API
+ * \author Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ *
+ */
+
+#include <sof/audio/codec_adapter/codec/generic.h>
+#include <sof/audio/codec_adapter/interfaces.h>
+
+/*****************************************************************************/
+/* Local helper functions						     */
+/*****************************************************************************/
+static int validate_config(struct codec_config *cfg);
+
+int
+codec_load_config(struct comp_dev *dev, void *cfg, size_t size,
+		  enum codec_cfg_type type)
+{
+	int ret;
+	struct codec_config *dst;
+	struct comp_data *cd = comp_get_drvdata(dev);
+	struct codec_data *codec = &cd->codec;
+
+	comp_dbg(dev, "codec_load_config() start");
+
+	if (!dev || !cfg || !size) {
+		comp_err(dev, "codec_load_config(): wrong input params! dev %x, cfg %x size %d",
+			 (uint32_t)dev, (uint32_t)cfg, size);
+		return -EINVAL;
+	}
+
+	dst = (type == CODEC_CFG_SETUP) ? &codec->s_cfg :
+					  &codec->r_cfg;
+
+	if (!dst->data) {
+		dst->data = rballoc(0, SOF_MEM_CAPS_RAM, size);
+	} else if (dst->size != size) {
+		rfree(dst->data);
```

Comment:
what is the purpose of rfree here?

Reply:
ack, this could do with more inline comments.